core/src/main/scala/org/http4s/Method.scala

```diff
@@ -1,104 +1,115 @@
 package org.http4s
 
-import scala.collection.concurrent.TrieMap
+import org.http4s.parser.HttpParser
+import org.http4s.util.{Writer, Renderable}
 import scalaz._
 
-import org.http4s.parser.Rfc2616BasicRules
-import org.http4s.util.{Writer, Renderable}
+/** The request method token is the primary source of request semantics; it
+  * indicates the purpose for which the client has made this request and what is
+  * expected by the client as a successful result.The method of an HTTP request.
+  *
+  * @see [http://tools.ietf.org/html/rfc7231#section-4 RFC7321, Section 4]
+  * @see [http://www.iana.org/assignments/http-methods/http-methods.xhtml IANA HTTP Method Registry]
+  */
+sealed abstract case class Method private (name: String) extends Renderable {
+  final override def render(writer: Writer): writer.type = writer << name
 
-import Method.Semantics
+  /** Request methods are considered "safe" if their defined semantics are
+    * essentially read-only
+    *
+    * @see https://tools.ietf.org/html/rfc7231#section-4.2.1
+    */
+  def isSafe: Boolean
 
-/**
- * An HTTP method.
- *
- * @see [http://tools.ietf.org/html/rfc7231#section-4 RFC7321, Section 4]
- * @see [http://www.iana.org/assignments/http-methods/http-methods.xhtml IANA HTTP Method Registry]
- */
-sealed abstract case class Method private (name: String) extends Renderable with Semantics {
-  final override def render(writer: Writer): writer.type = writer << name
+  /** A request method is considered "idempotent" if the intended effect on the
+    * server of multiple identical requests with that method is the same as the
+    * effect for a single such request.
+    *
+    * @see https://tools.ietf.org/html/rfc7231#section-4.2.2
+    */
+  def isIdempotent: Boolean
+
+  /** Request methods can be defined as "cacheable" to indicate that responses to
+   *  them are allowed to be stored for future reuse
+   *
+   *  @see https://tools.ietf.org/html/rfc7231#section-4.2.3
+   */
+  def isCacheable: Boolean
 }
 
 object Method extends MethodInstances {
-  sealed trait Semantics {
-    def isIdempotent: Boolean
-    def isSafe: Boolean
-  }
-
-  object Semantics {
-    trait Default extends Semantics {
-      def isIdempotent: Boolean = false
-      def isSafe: Boolean = false
-    }
-    trait Idempotent extends Semantics {
-      def isIdempotent: Boolean = true
-      def isSafe: Boolean = false
-    }
-    trait Safe extends Semantics {
-      def isIdempotent: Boolean = true
-      def isSafe: Boolean = true
-    }
-  }
-
   // Type tags for a method allowing a body or not
   sealed trait PermitsBody extends Method
```

Comment:
Because those marker types are used to drive what syntax is available in the client.  Which is a detail I don't like making part of core, but then we have to do something horrid like we did for Status syntax in DSL.  I'd love to hear a better idea here.